freeswitch_click2dial/freeswitch_click2dial.py

```diff
@@ -0,0 +1,416 @@
+    def _check_validity(self):
+        for server in self:
+            out_prefix = ('Out prefix', server.out_prefix)
+            dialplan_context = ('Dialplan context', server.context)
+            alert_info = ('Alert-Info SIP header', server.alert_info)
+            password = ('Event Socket password', server.password)
+
+            if out_prefix[1] and not out_prefix[1].isdigit():
+                raise ValidationError(
+                    _("Only use digits for the '%s' on the FreeSWITCH server "
+                        "'%s'" % (out_prefix[0], server.name)))
+            if server.wait_time < 1 or server.wait_time > 120:
+                raise ValidationError(
+                    _("You should set a 'Wait time' value between 1 and 120 "
+                        "seconds for the FreeSWITCH server '%s'"
+                        % server.name))
+            if server.port > 65535 or server.port < 1:
+                raise ValidationError(
+                    _("You should set a TCP port between 1 and 65535 for the "
+                        "FreeSWITCH server '%s'" % server.name))
+            for check_str in [dialplan_context, alert_info, password]:
+                if check_str[1]:
+                    try:
+                        check_str[1].encode('ascii')
+                    except UnicodeEncodeError:
+                        raise ValidationError(
+                            _("The '%s' should only have ASCII characters for "
+                                "the FreeSWITCH server '%s'"
+                                % (check_str[0], server.name)))
+
+    @api.model
+    def _connect_to_freeswitch(self):
+        '''
+        Open the connection to the FreeSWITCH Event Socket
+        Returns an instance of the FreeSWITCH Event Socket
+
+        '''
+        user = self.env.user
+
+        fs_server = user.get_freeswitch_server_from_user()
+        # We check if the current user has a chan type
+        if not user.freeswitch_chan_type:
+            raise UserError(
+                _('No channel type configured for the current user.'))
+
+        # We check if the current user has an internal number
+        if not user.resource:
+            raise UserError(
+                _('No resource name configured for the current user.'))
+
+        _logger.debug(
+            "User's phone: %s/%s", user.freeswitch_chan_type, user.resource)
+        _logger.debug(
+            "FreeSWITCH server: %s:%d", fs_server.ip_address, fs_server.port)
+
+        # Connect to the FreeSWITCH Event Socket
+        try:
+            fs_manager = ESL.ESLconnection(
+                str(fs_server.ip_address),
+                str(fs_server.port), str(fs_server.password))
+        except Exception, e:
+            _logger.error(
+                "Error in the request to the FreeSWITCH Event Socket %s",
+                fs_server.ip_address)
+            _logger.error("Here is the error message: %s", e)
```

Comment:
No need to call `_logger.error` second time, just add parametr `exc_info=True` to previous call